pkg/controller/workloadspread/workloadspread_controller.go

```diff
@@ -411,17 +431,23 @@ func (r *ReconcileWorkloadSpread) getSuitableSubsetForOldPod(ws *appsv1alpha1.Wo
 			klog.Errorf("unexpected error occurred when matching pod (%s/%s) with subset, please check requiredSelectorTerm field of subset (%s) in WorkloadSpread (%s/%s), err: %s",
 				pod.Namespace, pod.Name, subset.Name, ws.Namespace, ws.Name, err.Error())
 		}
+		quotaScore := int64(0)
+		if subsetMissingReplicas[subset.Name] > 0 {
+			quotaScore = int64(1)
+		}
+		finalScore := preferredScore*10 + quotaScore
 		// select the most favorite subsets for the pod by subset.PreferredNodeSelectorTerms
-		if matched && preferredScore > maxPreferredScore {
+		if matched && finalScore > maxPreferredScore {
 			favoriteSubset = subset
-			maxPreferredScore = preferredScore
+			maxPreferredScore = finalScore
 		}
 	}
 
 	if favoriteSubset != nil {
 		if err := r.patchFavoriteSubsetMetadataToPod(pod, ws, favoriteSubset); err != nil {
 			return nil, err
 		}
+		subsetMissingReplicas[favoriteSubset.Name]--
```

Comment:
can we patch metadata to pod outside getSuitableSubsetForOldPod, function of name getXXX seems readonly, it may surprise readers. Alternatively, we can rename getSuitableSubsetNameForPod to getAndUpdateSuitableSubsetName